File: golang/9_ticket_template/ticket.go
package __ticket_template

type Ticket struct {
	Id                  string               `json:"id"`          // ticket system id in ticket database
	KeyName             string               `json:"key_name"`    // unique identification of ticket system
	Admins              []string             `json:"admins"`      // the administrators of this ticket
	Url                 string               `json:"url"`         // url of this ticket system that I request to
	Desc                string               `json:"desc"`        // description of ticket system
	Homepage            string               `json:"homepage"`    // homepage of ticket system
	HttpMethod          string               `json:"http_method"` // method for ticket request
	Auth                AuthInfo             `json:"auth"`        //  auth information for ticket request
	SiteKeyAndEndpoints []SiteKeyAndEndpoint `json:"site_key_and_endpoints"`
	RequestParams       RequestParams        `json:"request_params"` // request_params request params of ticket system
	Status              string               `json:"status"`         // status of this ticket system, like "normal" "deleted"
}

type AuthInfo struct {
	Method string `json:"method"` // type of auth, like "basic/bear token"...
	Token  string `json:"token"`  // auth token for request
}

type SiteKeyAndEndpoint struct {
	SiteKey  string `json:"site_key"` // not need for requests
	Endpoint string `json:"endpoint"` // not need for requests
	IDC      string `json:"idc"`      // not need for requests
}

type RequestParams struct {
	ParamGroup []Param `json:"param_group" bson:"param_group"`
}

type Param struct {
	KeyName     string   `json:"key_name" bson:"key_name" vd:"keyName($)"`                              // key name stored in database
	Value       string   `json:"value" bson:"value"`                                                    // param value
	DisplayName string   `json:"display_name" bson:"display_name"`                                      // key display name
	ElementType string   `json:"element_type" bson:"element_type" vd:"in($, 'int', 'float', 'string')"` // key type, e.g. int, float, string
	Structure   string   `json:"structure" bson:"structure" vd:"in($, 'single', 'list')"`               // array or not array, e.g. single, list
	Required    bool     `json:"required" bson:"required"`                                              // whether this param is required
	Editable    bool     `json:"editable" bson:"editable"`                                              // whether this param is editable
	Options     []string `json:"options" bson:"options"`                                                // option value of this param
}

File: myjava/src/main/java/org/example/tickettemplate/Ticket.java
package org.example.tickettemplate;

import java.util.List;

public class Ticket {
    private String id;
    private String keyName;
    private List<String> admins;
    private String url;
    private String desc;
    private String homepage;
    private String httpMethod;
    private AuthInfo auth;
    private List<SiteKeyAndEndpoint> siteKeyAndEndpoints;
    private RequestParams requestParams;
    private String status;

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getKeyName() { return keyName; }
    public void setKeyName(String keyName) { this.keyName = keyName; }

    public List<String> getAdmins() { return admins; }
    public void setAdmins(List<String> admins) { this.admins = admins; }

    public String getUrl() { return url; }
    public void setUrl(String url) { this.url = url; }

    public String getDesc() { return desc; }
    public void setDesc(String desc) { this.desc = desc; }

    public String getHomepage() { return homepage; }
    public void setHomepage(String homepage) { this.homepage = homepage; }

    public String getHttpMethod() { return httpMethod; }
    public void setHttpMethod(String httpMethod) { this.httpMethod = httpMethod; }

    public AuthInfo getAuth() { return auth; }
    public void setAuth(AuthInfo auth) { this.auth = auth; }

    public List<SiteKeyAndEndpoint> getSiteKeyAndEndpoints() { return siteKeyAndEndpoints; }
    public void setSiteKeyAndEndpoints(List<SiteKeyAndEndpoint> siteKeyAndEndpoints) { this.siteKeyAndEndpoints = siteKeyAndEndpoints; }

    public RequestParams getRequestParams() { return requestParams; }
    public void setRequestParams(RequestParams requestParams) { this.requestParams = requestParams; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
}

class AuthInfo {
    private String method;
    private String token;

    public String getMethod() { return method; }
    public void setMethod(String method) { this.method = method; }

    public String getToken() { return token; }
    public void setToken(String token) { this.token = token; }
}

class SiteKeyAndEndpoint {
    private String siteKey;
    private String endpoint;
    private String idc;

    public String getSiteKey() { return siteKey; }
    public void setSiteKey(String siteKey) { this.siteKey = siteKey; }

    public String getEndpoint() { return endpoint; }
    public void setEndpoint(String endpoint) { this.endpoint = endpoint; }

    public String getIdc() { return idc; }
    public void setIdc(String idc) { this.idc = idc; }
}

class RequestParams {
    private List<Param> paramGroup;

    public List<Param> getParamGroup() { return paramGroup; }
    public void setParamGroup(List<Param> paramGroup) { this.paramGroup = paramGroup; }
}

class Param {
    private String keyName;
    private String value;
    private String displayName;
    private String elementType;
    private String structure;
    private boolean required;
    private boolean editable;
    private List<String> options;

    public String getKeyName() { return keyName; }
    public void setKeyName(String keyName) { this.keyName = keyName; }

    public String getValue() { return value; }
    public void setValue(String value) { this.value = value; }

    public String getDisplayName() { return displayName; }
    public void setDisplayName(String displayName) { this.displayName = displayName; }

    public String getElementType() { return elementType; }
    public void setElementType(String elementType) { this.elementType = elementType; }

    public String getStructure() { return structure; }
    public void setStructure(String structure) { this.structure = structure; }

    public boolean isRequired() { return required; }
    public void setRequired(boolean required) { this.required = required; }

    public boolean isEditable() { return editable; }
    public void setEditable(boolean editable) { this.editable = editable; }

    public List<String> getOptions() { return options; }
    public void setOptions(List<String> options) { this.options = options; }
}
